tools/azure-pipelines-tasks-json-generate.mjs

```diff
@@ -0,0 +1,70 @@
+import os from 'os';
+import { promisify } from 'util';
+import fs from 'fs-extra';
+import g from 'glob';
+import JSON5 from 'json5';
+import shell from 'shelljs';
+import Git from 'simple-git';
+import path from 'upath';
+
+const glob = promisify(g);
+const localPath = path.join(os.tmpdir(), 'azure-pipelines-tasks');
+
+/**
+ * This script:
+ *  1. Clones the Azure Pipelines Tasks repo
+ *  2. Finds all `task.json` files
+ *  3. For each `task.json` it finds each commit that has that file
+ *  4. For each commit it gets the `task.json` content and extracts the task name and version
+ *  5. After all the `task.json` files have been processed it writes the results to `./data/azure-pipelines-tasks.json`
+ */
+await (async () => {
+  await fs.ensureDir(localPath);
+  const git = Git(localPath);
+
+  if (await git.checkIsRepo()) {
+    await git.pull();
+  } else {
+    await git.clone(
+      'https://github.com/microsoft/azure-pipelines-tasks.git',
+      '.'
+    );
+  }
+
+  // Find all `task.json` files
+  const files = (await glob(path.join(localPath, '**/task.json'))).map((file) =>
+    file.replace(`${localPath}/`, '')
+  );
+
+  /** @type {Record<string, Set<string>>} */
+  const tasks = {};
+
+  for (const file of files) {
+    // Find all commits that have the file
+    const revs = (await git.raw(['rev-list', 'HEAD', '--', file])).split('\n');
+    shell.echo(`Parsing ${file}`);
+    for (const rev of revs) {
+      try {
+        // Get the content of the file at the commit
+        const content = await git.show([`${rev}:${file}`]);
+        /** @type {{name: string, version: {Major: number, Minor: number, Patch: number}}} */
+        const parsedContent = JSON5.parse(content);
+        const version = `${parsedContent.version.Major}.${parsedContent.version.Minor}.${parsedContent.version.Patch}`;
+        tasks[parsedContent.name.toLowerCase()] =
```

Comment:
i think we should use upper case, as there are chars which doesn't have a lower case variant

Reply:
In general I agree, but task names are restricted to `^[A-Za-z0-9_]+$`. So in this specific instance I think we are okay.

Reply:
In general I agree, but task names are restricted to `^[A-Za-z0-9_]+$`. So in this specific instance I think we are okay.